codebase/billing-service-secondary-async/src/main/scala/com/virtuslab/billings/async/S3Service.scala

```diff
@@ -0,0 +1,36 @@
+package com.virtuslab.billings.async
+
+import java.time.LocalDateTime
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+
+import akka.actor.ActorSystem
+import akka.stream.Materializer
+import akka.stream.alpakka.s3.S3Settings
+import akka.stream.alpakka.s3.scaladsl.{MultipartUploadResult, S3Client}
+import akka.stream.scaladsl.{Sink, Source}
+import akka.util.ByteString
+import com.virtuslab.payments.payments.PaymentRequest
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+import scala.util.Random
+
+trait S3Service {
+
+  protected implicit def system: ActorSystem
+  protected implicit val materializer: Materializer
+
+  private val bucket = "reactsphere-billing-service-data"
+  protected lazy val s3Client = new S3Client(S3Settings())
+
+  def putInvoice(data: PaymentRequest): Future[String] = {
+    val timestamp = LocalDateTime.now().format(DateTimeFormatter.BASIC_ISO_DATE)
```

Comment:
I think it might make sense to use actual PDF here. I was thinking about something along the lines of:
- having dummy PDF file in resources (say something of a size of several hundreds kB - 200-400 kB)
- uploading it to S3 (streaming from disk)

Reply:
Done. Best invoice you've ever seen:
https://github.com/VirtusLab/ReactSphere-reactive-beyond-hype/blob/e0097e95effbbb61ac48923c2480069d536aa360/codebase/commons/src/main/resources/dummy_invoice.pdf